test_pytest_rerunfailures.py

```diff
@@ -219,3 +221,32 @@ def test_fail():
                                '--result-log', './pytest.log')
 
     assert_outcomes(result, passed=0, failed=1, rerun=2)
+
+
+@pytest.mark.parametrize('delay_time', [0, 0.0, 1, 2.5])
+def test_reruns_with_delay(testdir, delay_time):
+    testdir.makepyfile("""
+        def test_fail():
+            assert False""")
+
+    result = testdir.runpytest('--reruns', '3',
+                               '--reruns-delay', delay_time)
+
+    assert_outcomes(result, passed=0, failed=1, rerun=3)
+
+
+def test_reruns_with_delay_marker(testdir):
+    testdir.makepyfile("""
+        import pytest
+        
+        @pytest.mark.flaky(reruns_delay=1)
+        def test_fail_one():
+            assert False
+            
+        @pytest.mark.flaky(reruns=2, reruns_delay=1)
+        def test_fail_two():
+            assert False""")
+
+    result = testdir.runpytest()
+
+    assert_outcomes(result, passed=0, failed=2, rerun=3)
```

Comment:
So these tests are good to check nothing blows up and the tests are run the correct number of times, but do we need a check that the delay is actually getting put in place?

Did you consider we could mock out time.sleep, and assert it's called with the correct value? This would have the side benefit of speeding up the test run - not that that really matters here :-)

Reply:
Yes, that a good idea. For this should I just use `unittest.mock` or is there a way to do it with the `monkeypatch` fixture?

Reply:
That sounds like a good idea, +1.

Reply:
For Python3 I will use `unittest.mock` but for Python2 this requires the `mock` package to be installed. Since this package is only required for the test I don't see any point in adding it to `setup.py`.  Is it ok if I just add it as a dependency in `tox.ini`?